piper/window.py

```diff
@@ -62,6 +62,20 @@ def __init__(self, ratbag, *args, **kwargs):
         else:
             self._present_welcome_perspective(ratbag.devices)
 
+    def do_delete_event(self, event):
+        for perspective in self.stack_perspectives.get_children():
+            if not perspective.can_shutdown:
+                dialog = Gtk.MessageDialog(self, Gtk.DialogFlags.MODAL,
+                                           Gtk.MessageType.QUESTION,
+                                           Gtk.ButtonsType.YES_NO,
+                                           _("There are unsaved changes. Are you sure you want to quit?"))
```

Comment:
if we use "apply" for the button we should use "unapplied" here too instead of unsaved.